### ISSUE TITLE
Adding multiple columns in SQL Server migrations fails

### ISSUE THREAD

**Issue by nickwhiteley** [knex/knex#1348]
The syntax generated for mssql, when adding a column to an existing table in a migration file, prefixes every column with ADD.  Only the first column added should have the ADD clause, others are column separated.

``` js
knex.schema.table('users', function (table) {
  table.string('first_name');
  table.string('last_name');
})
```

Generates:

``` sql
ALTER TABLE [users] ADD  [first_name] nvarchar(255), ADD [last_name] nvarchar(255)
```

But it should generate:

``` sql
ALTER TABLE [users] ADD  [first_name] nvarchar(255), [last_name] nvarchar(255)
```

or:

``` sql
ALTER TABLE [users] ADD  [first_name] nvarchar(255)
ALTER TABLE [users] ADD  [last_name] nvarchar(255)
```

Adding the following to the assign in dialects\mssql\schema\tablecompiler.js fixed it for me.

``` js
  // Add columns to the table
  addColumns : function (columns) {
    var addClause = ' ';
    for (var i = 0, l = columns.sql.length; i < l; i++) {
      this.pushQuery({
        sql: 'ALTER TABLE ' + this.tableName() + ' ADD ' + columns.sql[i],
        bindings: columns.bindings[i]
      })
    };
  },

```

Putting it in a single ALTER statement breaks the rollback (the way I did it, can't see why) hence opting for multiple ALTER statements.

**Comment by rhys-vdw** [comment]
@nickwhiteley Thanks for the report. Could you provide a PR for this change please?

**Comment by dhensby** [comment]
It looks like this issue can be closed as it was fixed in #1348?

**Comment by kibertoad** [comment]
@dhensby Do you mean #1401?

**Comment by nickwhiteley** [comment]
Fixed in PR #1401
